packages/bitcore-wallet-service/src/lib/server.ts

```diff
@@ -1778,67 +1779,7 @@ export class WalletService {
 
       this.syncWallet(wallet, err => {
         if (err) return cb(err);
-
-        if (! wallet.isUTXOCoin()) {
-          bc.getBalance(wallet, (err, balance) => {
-            if (err) {
-              return cb(err);
-            }
-            this.getPendingTxs({}, (err, txps) => {
-              if (err) return cb(err);
-              const lockedSum = _.sumBy(txps, 'amount');
-              const convertedBalance = this._convertBitcoreBalance(balance, lockedSum);
-              this.storage.fetchAddresses(this.walletId, (
-                err,
-                addresses: IAddress[]
-              ) => {
-                if (err) return cb(err);
-                if (addresses.length > 0) {
-                  const byAddress = [{
-                    address: addresses[0].address,
-                    path: Constants.PATHS.SINGLE_ADDRESS,
-                    amount: convertedBalance.totalAmount
-                  }];
-                  convertedBalance.byAddress = byAddress;
-                }
-                return cb(null, convertedBalance);
-              });
-            });
-          });
-        } else {
-          this._getUtxosForCurrentWallet(
-            {
-              coin: opts.coin,
-              addresses: opts.addresses
-            },
-            (err, utxos) => {
-              if (err) return cb(err);
-
-              const balance = { ...this._totalizeUtxos(utxos), byAddress: [] };
-
-              // Compute balance by address
-              const byAddress = {};
-              _.each(_.keyBy(_.sortBy(utxos, 'address'), 'address'), (
-                value,
-                key
-              ) => {
-                byAddress[key] = {
-                  address: key,
-                  path: value.path,
-                  amount: 0
-                };
-              });
-
-              _.each(utxos, (utxo) => {
-                byAddress[utxo.address].amount += utxo.satoshis;
-              });
-
-              balance.byAddress = _.values(byAddress);
-
-              return cb(null, balance);
-            }
-          );
-        }
+        return ChainService.getWalletBalance(opts, cb);
```

Comment:
not passing wallet in here, but the method loads from wallet.coin

```
return ChainService.getWalletBalance(opts, cb);
```

Types would have caught this one